api/grpc/Server.go

```diff
@@ -155,7 +155,8 @@ func (s *Server) Serve(wait *sync.WaitGroup) error {
 			s.ServerConfig.GrpcPort)
 	}
 
-	authMiddleware, err := interceptor.NewAuthnInterceptor(s.ServerConfig.AuthConfig)
+	authConfigs := []api.AuthConfig{s.ServerConfig.AuthConfig}
```

Comment:
Just a loose thought for now: thinking in term of yaml config here, if we go further down this road (which I have to think about but looks good at first glance), I think the to-be-done yaml config may look like
```
app:
    auth:
        id: "stage-sso"
            discoveryEndpoint: ...
            audience: ...
            minimalScope: ...
        id: "prod-sso"
            discoveryEndpoint: ...
            audience: ...
            minimalScope: ...
```

Reply:
Looks about right to me. :)